tests/app/data_model/test_metadata_proxy.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from werkzeug.datastructures import ImmutableDict
+
+from app.data_models.metadata_proxy import MetadataProxy, SurveyMetadata
+
+METADATA_V1 = {
+    "schema_name": "1_0000",
+    "ru_ref": "432423423423",
+    "response_id": "1",
+    "account_service_url": "account_service_url",
+    "tx_id": "tx_id",
+    "collection_exercise_sid": "collection_exercise_sid",
+    "case_id": "case_id",
+}
+
+METADATA_V2 = {
+    "version": "v2",
+    "schema_name": "1_0000",
+    "response_id": "1",
+    "account_service_url": "account_service_url",
+    "tx_id": "tx_id",
+    "collection_exercise_sid": "collection_exercise_sid",
+    "case_id": "case_id",
+    "survey_metadata": {
+        "data": {
+            "ru_ref": "432423423423",
+        },
+    },
+}
+
+
+@pytest.mark.parametrize(
+    "resolved_metadata_proxy_value, metadata_var",
+    (
+        (MetadataProxy.from_dict(dict(METADATA_V1))["ru_ref"], METADATA_V1["ru_ref"]),
```

Comment:
This could be simplified, but won't block, however, these `dict()` calls are redundant?